### ISSUE TITLE
azurerm_maintenance_assignment_virtual_machine segfaults

### ISSUE THREAD

**Issue by pierluca** [hashicorp/terraform-provider-azurerm#20780]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.9

### AzureRM Provider Version

3.44.1

### Affected Resource(s)/Data Source(s)

azurerm_maintenance_assignment_virtual_machine

### Terraform Configuration Files

```hcl
resource "azurerm_maintenance_configuration" "example" {
  name                = "${var.vm_name}-mc"
  resource_group_name = var.resource_group_name
  location            = var.location
  scope               = "InGuestPatch"

  window {
    start_date_time = var.maintenance_start_date_time
    duration        = var.maintenance_duration
    recur_every     = var.maintenance_recur_every
    time_zone       = "W. Europe Standard Time"
  }

  # Needed if scope is "InGuestPatch"
  install_patches {
    linux {
      classifications_to_include = [
        "Critical",
        "Security",
        "Other"
      ]
      package_names_mask_to_exclude = []
      package_names_mask_to_include = []
    }
    reboot = "IfRequired"
  }

  # Needed if scope is "InGuestPatch"
  in_guest_user_patch_mode = "User"
}

resource "azurerm_maintenance_assignment_virtual_machine" "this" {
  location                     = var.location
  maintenance_configuration_id = azurerm_maintenance_configuration.example.id
  virtual_machine_id           = azurerm_linux_virtual_machine.this.id
}
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.44.1_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x55c89b3]

goroutine 913 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance.getMaintenanceAssignmentVirtualMachine({0x74ff428?, 0xc001c33080?}, 0x0?, 0x0?, {0xc0022f2e10, 0x96})
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance/maintenance_assignment_virtual_machine_resource.go:194 +0x253
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance.resourceArmMaintenanceAssignmentVirtualMachineRead(0xc001f66f80, {0x615acc0?, 0xc002580c00?})
github.com/hashicorp/terraform-provider-azurerm/internal/services/maintenance/maintenance_assignment_virtual_machine_resource.go:132 +0x18d
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x74ff460?, {0x74ff460?, 0xc000613800?}, 0xd?, {0x615acc0?, 0xc002580c00?})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:712 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0xc000dbd880, {0x74ff460, 0xc000613800}, 0xc0025eb930, {0x615acc0, 0xc002580c00})
github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/resource.go:1015 +0x585
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0xc0005b9998, {0x74ff460?, 0xc000831770?}, 0xc002893240)
github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.1/helper/schema/grpc_provider.go:613 +0x4a5
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0xc000a634a0, {0x74ff460?, 0xc000afcf90?}, 0xc001c32e40)
github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/tf5server/server.go:748 +0x4b1
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x6879fa0?, 0xc000a634a0}, {0x74ff460, 0xc000afcf90}, 0xc0008c6d20, 0x0)
github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00001e000, {0x750f940, 0xc0011e6ea0}, 0xc001f4cea0, 0xc0013e1650, 0xb9dd650, 0x0)
google.golang.org/grpc@v1.51.0/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc00001e000, {0x750f940, 0xc0011e6ea0}, 0xc001f4cea0, 0x0)
google.golang.org/grpc@v1.51.0/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
google.golang.org/grpc@v1.51.0/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
google.golang.org/grpc@v1.51.0/server.go:963 +0x28a

Error: The terraform-provider-azurerm_v3.44.1_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behaviour

A working deployment

### Actual Behaviour

```
Error: Plugin did not respond
[...]
The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
```

### Steps to Reproduce

I'm still working on a minimal repro, as I've bumped into cases where this actually works.

### References

The discussion around this issue started in https://github.com/hashicorp/terraform-provider-azurerm/issues/20125

**Comment by pierluca** [comment]
@lonegunmanb the minimal repro I wrote, inconveniently, doesn't crash. Some repros where I have this logic work, and others crash. It's a bit weird and I'm still trying to figure out how to trigger the crash.

Based on your preliminary fix, do you have any clue as to what might cause it ?

**Comment by lonegunmanb** [comment]
Sure thing @pierluca, I'll try to reproduce the panic, no matter what the result is I'll submit my pr very soon.

**Comment by pierluca** [comment]
Awesome, thank you @lonegunmanb !
I'm still working on the minimal repro, I'll keep you posted if there's a breakthrough.

**Comment by lonegunmanb** [comment]
Hi @pierluca, I've submitted my pr though I was failed to reproduce this panic on my side, I hope the patch would solve your issue, please stay tuned until this pr has been merged and a new version has been released. Thanks for your understanding.

**Comment by rcskosir** [comment]
Thanks for taking the time to submit this issue. I am going to mark this issue as closed with #20781. If that is not the case, please provide additional information including the version in which you are still experiencing this issue, thanks!
